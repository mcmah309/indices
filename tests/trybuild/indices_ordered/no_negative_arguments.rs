use indices::indices_ordered;

fn main() {
    let mut data = [5,4,3,2,1];
    let slice = data.as_mut_slice();
    let (_one, _two) = indices_ordered!(slice, -1, 1);
}