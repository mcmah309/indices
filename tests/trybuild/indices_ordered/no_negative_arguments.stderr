error[E0277]: the trait bound `usize: Neg` is not satisfied
 --> tests/trybuild/indices_ordered/no_negative_arguments.rs:6:48
  |
6 |     let (_one, _two) = indices_ordered!(slice, -1, 1);
  |                                                ^^ the trait `Neg` is not implemented for `usize`
  |
  = help: the following other types implement trait `Neg`:
            &f128
            &f16
            &f32
            &f64
            &i128
            &i16
            &i32
            &i64
          and $N others

error[E0600]: cannot apply unary operator `-` to type `usize`
 --> tests/trybuild/indices_ordered/no_negative_arguments.rs:6:48
  |
6 |     let (_one, _two) = indices_ordered!(slice, -1, 1);
  |                                                ^^
  |                                                |
  |                                                cannot apply unary operator `-`
  |                                                help: you may have meant the maximum value of `usize`: `usize::MAX`
  |
  = note: unsigned values cannot be negated
